samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -182,4 +183,30 @@ public void handleDataDeleted(String dataPath)
       LOG.warn("barrier done got deleted at " + dataPath);
     }
   }
+
+  class BarrierKeyBuilder {
```

Comment:
Yeah. I spent an hour debugging a unit test because there was a typo in one of the keys. I didn't notice and there wasn't a doc for the actual value. So, decided to just add it here for now.